MappingIT trips assertion and causes nodes to crash in 7.x bwc tests

While working on a backport, I found that an aspect of the rolling upgrade tests did not work properly so tests that checked the `UPGRADE_FROM_VERSION` value and executed only when upgrading from 6.x were never being executed. I'm fixing the issue with the rolling upgrade tests in #60927 and will mute MappingIT in the rolling upgrade qa project since it trips an assertion and causes nodes to die.

```
» fatal error in thread [elasticsearch[v6.8.12-0][clusterApplierService#updateTask][T#1]], exiting
»  java.lang.AssertionError: expected current mapping [{"_doc":{"_all":{"enabled":false},"properties":{"field":{"type":"text"}}}}] for type [_doc] to be the same as new mapping [{"_doc":{"properties":{"field":{"type":"text"}}}}]
»       at org.elasticsearch.index.mapper.MapperService.assertMappingVersion(MapperService.java:304)
»       at org.elasticsearch.index.mapper.MapperService.updateMapping(MapperService.java:215)
»       at org.elasticsearch.index.IndexService.updateMapping(IndexService.java:631)
»       at org.elasticsearch.indices.cluster.IndicesClusterStateService.updateIndices(IndicesClusterStateService.java:530)
»       at org.elasticsearch.indices.cluster.IndicesClusterStateService.applyClusterState(IndicesClusterStateService.java:244)
»       at org.elasticsearch.cluster.service.ClusterApplierService.lambda$callClusterStateAppliers$5(ClusterApplierService.java:522)
»       at java.base/java.lang.Iterable.forEach(Iterable.java:75)
»       at org.elasticsearch.cluster.service.ClusterApplierService.callClusterStateAppliers(ClusterApplierService.java:519)
»       at org.elasticsearch.cluster.service.ClusterApplierService.applyChanges(ClusterApplierService.java:490)
»       at org.elasticsearch.cluster.service.ClusterApplierService.runTask(ClusterApplierService.java:437)
»       at org.elasticsearch.cluster.service.ClusterApplierService.access$100(ClusterApplierService.java:74)
»       at org.elasticsearch.cluster.service.ClusterApplierService$UpdateTask.run(ClusterApplierService.java:177)
»       at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:651)
»       at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.runAndClean(PrioritizedEsThreadPoolExecutor.java:252)
»       at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.run(PrioritizedEsThreadPoolExecutor.java:215)
»       at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)
»       at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)
»       at java.base/java.lang.Thread.run(Thread.java:832)
```

The fix to the rolling-upgrade build that provokes this is https://github.com/elastic/elasticsearch/pull/60927/commits/d05588a56f5167a5170bbcfbc5d3cc21bdb03e46.

cc @romseygeek since I see your name in git for this assertion.

--- a/qa/rolling-upgrade/src/test/java/org/elasticsearch/upgrades/MappingIT.java
+++ b/qa/rolling-upgrade/src/test/java/org/elasticsearch/upgrades/MappingIT.java
@@ -18,13 +18,11 @@
  */
 package org.elasticsearch.upgrades;
 
-import org.apache.lucene.util.LuceneTestCase.AwaitsFix;
 import org.elasticsearch.Version;
 import org.elasticsearch.client.Request;
 import org.elasticsearch.client.Response;
 import org.elasticsearch.common.xcontent.support.XContentMapValues;
 
-@AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/60986")
 public class MappingIT extends AbstractRollingTestCase {
     /**
      * Create a mapping that explicitly disables the _all field (possible in 6x, see #37429)

--- a/server/src/main/java/org/elasticsearch/index/mapper/AllFieldMapper.java
+++ b/server/src/main/java/org/elasticsearch/index/mapper/AllFieldMapper.java
@@ -77,7 +77,7 @@ public class AllFieldMapper extends MetadataFieldMapper {
     }
 
     public static final TypeParser PARSER = new ConfigurableTypeParser(
-        c -> new AllFieldMapper(new Explicit<>(true, false)),
+        c -> new AllFieldMapper(new Explicit<>(false, false)),
         c -> new Builder()
     );
 
@@ -128,4 +128,8 @@ public class AllFieldMapper extends MetadataFieldMapper {
         return CONTENT_TYPE;
     }
 
+    @Override
+    public ParametrizedFieldMapper.Builder getMergeBuilder() {
+        return new Builder().init(this);
+    }
 }

--- a/server/src/test/java/org/elasticsearch/index/mapper/AllFieldMapperTests.java
+++ b/server/src/test/java/org/elasticsearch/index/mapper/AllFieldMapperTests.java
@@ -54,6 +54,8 @@ public class AllFieldMapperTests extends ESSingleNodeTestCase {
                 .endObject().endObject()
             );
             indexService.mapperService().merge("_doc", new CompressedXContent(mappingDisabled), MergeReason.MAPPING_UPDATE);
+            assertEquals("{\"_doc\":{\"_all\":{\"enabled\":false}}}",
+                Strings.toString(indexService.mapperService().documentMapper()));
 
             String mappingEnabled = Strings.toString(XContentFactory.jsonBuilder().startObject()
                 .startObject("_all")

